### PR TITLE
feat(model): add portholes

### LINKED ISSUE #6094: Add more hublots
### Aircraft Version

Development

### Description

Add more hublots to the model. Two opposite the existing ones, and two under the wing.

### References (optional)

_No response_

### Additional info (optional)

I'm working on it.

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by igor8518**
Fixes #6094 

## Summary of Changes
This PR adds four portholes. Two opposite the existing ones, and two above the wing.

### Nodes for control:
**Legacy:**

-   HublotIn
-   glass
-   HublotIn002
-   glass002

**Added:**

-   HublotIn003
-   glass003
-   HublotIn004
-   glass004
-   HublotIn005
-   glass005
-   HublotIn006
-   glass006

 

Discord username: igor8518

## Testing instructions
Just check if there are any visual artifacts on the internal and external model.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by BlueberryKing** [comment]
Hello! If you are still working on this PR, you can mark it as a draft on GitHub to make it clear to everybody.

**Comment by bouveng** [comment]
@igor8518 Hi! Nice! What's left to do on this? Could you provide the node-names for all port-holes as additional context for this PR? Would be useful if we want to toggle the visibility for all portholes sometime in the future.

Br.

**Comment by igor8518** [comment]
> @igor8518 Hi! Nice! What's left to do on this? Could you provide the node-names for all port-holes as additional context for this PR? Would be useful if we want to toggle the visibility for all portholes sometime in the future.
> 
> Br.

Hi!
Done.

**Review by bouveng — APPROVED** [review]
LGTM, but @tracernz should have a look as well.

**Comment by marcman86** [comment]
Quality Assurance Tester Report
Discord:marcman86#4907
Object of testing: #6107
Tier of Testing: 1
Date: 30/10/2021

Testing Process:
- Spawned C&D at EDDB
- Swithed to fixed external views

Negatives: - Windows are only visible on external view 12 and 13

Testing Results: Not passed
Conclusions: Same as before the PR

Hope this helps :-)

**Comment by bouveng** [comment]
> Negatives: - Windows are only visible on external view 12 and 13

@marcman86 Are you using the default A32NX views? I think these views are part of the fixed showcase views.

Br.

**Comment by marcman86** [comment]
I tested all available fixed external views. Also the ones added by #3702. In these views i expect the new windows. If not then the PR description is not clear enough.

**Comment by bouveng** [comment]
> In these views i expect the new windows. If not then the PR description is not clear enough.

Sure, these should be in the same array as the 2 other porthole views. And should be visible when using the showcase cam.

@igor8518 Where are the windows? :see_no_evil:

**Comment by igor8518** [comment]
Corrected fixed cameras 14-19 in cameras.cfg. I also corrected screenshots from new cameras. I suggest removing cameras 12, 13.

**Review by bouveng — APPROVED** [review]

**Comment by frankkopp** [comment]
Quality Assurance Tester Report

Discord: Cdr_Maverick#6475
Object of testing: #6107
Tier of Testing: 1
Date: 31.10.2021

Testing Process: 
C&D at gate
External and Drone view detailed checks
Fixed camera checks
Looking for artefacts and errors
Tested with default livery and one custom livery

Negatives: none

Testing Results: passed

Conclusions: some redundancy but this has been commented already.

**Comment by marcman86** [comment]
Quality Assurance Tester Report
Discord:marcman86#4907
Object of testing: #6107
Tier of Testing: 1
Date: 31/10/2021

Testing Process:
- Spawned C&D at EDDV
- Swithed to fixed external views

Negatives: - 

Testing Results: Passed
Conclusions: Works for me

Hope this helps :-)

**Comment by tracernz** [comment]
So up to now we've only accepted valid gltf files for merging, and used models.json to modify/add individual bits. I see you've changed it to accept invalid/incomplete gltf files...  Can we not just do the same thing as we've done before even if just for consistencies sake?

**Comment by igor8518** [comment]
I agree with you. It just seems to me that models.json turns into an inconvenient file for editing.  

I'll redo it according to your conditions.

**Comment by bouveng** [comment]
@igor8518 Changed to draft until new impl. is ready.

**Comment by tracernz** [comment]
> I agree with you. It just seems to me that models.json turns into an inconvenient file for editing.
> 
> I'll redo it according to your conditions.

Yeah, I in some ways like what you did. The thing that swings me is consistency. Let me know if it proves difficult for some reason and we can reconsider.

**Comment by igor8518** [comment]
@tracernz, changed the method of adding nodes to the scene, according to your requirements.

**Review by bouveng — APPROVED** [review]
Needs approval from @tracernz as well.

**Review by tracernz — APPROVED** [review]
Thanks for the contribution!

I think the model script is getting to the point it needs a refactor (it really was prior to this PR as well 😛), but that is beyond the scope of your PR.

**Comment by igor8518** [comment]
Thank you, Michael. 
I'm currently working on a passenger cabin. At the same time, I found a lot of problems with the visual model from Asobo.

**Comment by tracernz** [comment]
> 
> 
> Thank you, Michael. I'm currently working on a passenger cabin. At the same time, I found a lot of problems with the visual model from Asobo.

Are you able to go on Discord? It would be good to loop in with our modelling people, as they're already working on an entirely new A320 model, using none of the Asobo assets. https://discord.gg/flybywire

**Comment by igor8518** [comment]
I'll try a little later.

**Comment by bouveng** [comment]
@igor8518 Please fix linting errors. 🔍🧐

```
/external/src/model/build.js
  50:9   error  'nodesBackup' is never reassigned. Use 'const' instead  prefer-const

  88:32  error  Block must not be padded by blank lines                 padded-blocks

  97:2   error  Unnecessary semicolon                                   no-extra-semi`
```

File: src/model/build.js
Comment (bouveng) [inline src/model/build.js:50]:
```suggestion
    const nodesBackup = nodes;
```

File: src/model/build.js
Comment (bouveng) [inline src/model/build.js:90]:
```suggestion
        gltfA.nodes.push(node);
    }
```

File: src/model/build.js
Comment (bouveng) [inline src/model/build.js:97]:
```suggestion
}
```

**Review by bouveng — CHANGES_REQUESTED** [review]
Fix lint errors, suggestions added.

```
50:9   error  'nodesBackup' is never reassigned. Use 'const' instead  prefer-const
88:32  error  Block must not be padded by blank lines                 padded-blocks
97:2   error  Unnecessary semicolon                                   no-extra-semi
```

**Review by bouveng — APPROVED** [review]

**Comment by marcman86** [comment]
Quality Assurance Tester Report
Discord:marcman86#4907
Object of testing: #6107
Tier of Testing: 1
Date: 6/11/2021

Testing Process:
- Spawned C&D at EDDH
- Swithed to fixed external views

Negatives: - 

Testing Results: Passed
Conclusions: Works for me

Hope this helps :-)

**Review by tracernz — APPROVED** [review]

**Comment by xZreai** [comment]
Quality Assurance Trainee Report

Discord: Cleon#5802
Object of testing: #6107
Tier of Testing: (1)
Date: (07/11/2021)

Testing Process: 
Spawned C/D at VHHH
Went into camera options and checked new camera views
Tested camera views via keybinds


Negatives: Nil


Testing Results: PASS


Conclusions: nil


Media: 
A nice departure on today's VATSIM event ;)
image

**Comment by igor8518** [comment]
@xZreai, thank you.))

**Comment by bouveng** [comment]
@igor8518 Merged! :sunglasses:

**Comment by bouveng** [comment]
> Corrected fixed cameras 14-19 in cameras.cfg. I also corrected screenshots from new cameras in EDDB(!) airport. I suggest removing cameras 12, 13.

Cameras 12, 13 where redundant, and now replaced with the previous views 14-15, that some user missed.
See https://github.com/flybywiresim/a32nx/pull/6158

**Comment by bouveng** [comment]
@igor8518 They seem to like it =) https://www.facebook.com/FlyByWireSimulations/posts/364343638770553

**Comment by igor8518** [comment]
:)

**Comment by igor8518** [comment]
How to insert a picture? )))

**Comment by igor8518** [comment]
I say right away, this is not my salon. I just implemented it with a few changes.

**Pull request merged**
